Tests/LagrangeTest/PolynomialTest.hpp

```diff
@@ -0,0 +1,119 @@
+/* GRTeclyn
+ * Copyright 2022 The GRTL collaboration.
+ * Please refer to LICENSE in GRTeclyn's root directory.
+ */
+
+#ifndef POLYNOMIALTEST_HPP_
+#define POLYNOMIALTEST_HPP_
+
+// AMReX includes
+#include <AMReX_AmrLevel.H>
+#include <AMReX_Geometry.H>
+#include <AMReX_MultiFab.H>
+#include "AMReX_Array.H"
+
+// My attempt to make a derived variable (a polynomial). I followed what has
+// been done for Constraints before.
+
+class PolynomialTest
+{
+  public:
+
+    static inline amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> my_center{
+        AMREX_D_DECL(0., 0., 0.)}; // some random default here
+
+    // read in the center which will be used to populating the polynomial
+    // properly
+    static void set_center(const std::array<double, AMREX_SPACEDIM> &c)
+    {
+        for (int d = 0; d < AMREX_SPACEDIM; ++d)
+            my_center[d] = c[d];
+    }
+
+    static inline const std::string name = "polynomial";
+
+    static inline const amrex::Vector<std::string> var_names = {"pol"};
+
+    // register with AMReX derive list
+    AMREX_FORCE_INLINE static void set_up(int a_state_index)
+    {
+        const auto &desc_lst = amrex::AmrLevel::get_desc_lst();
+        auto &derive_lst     = amrex::AmrLevel::get_derive_lst();
+
+        derive_lst.add(
+            name, amrex::IndexType::TheCellType(), 1, var_names, compute_mf,
+            // grow by 2 (ghost cells)
+            [](const amrex::Box &b) { return amrex::grow(b, 2); },
```

Comment:
I don't think you need ghosts for this "derived" quantity.
```suggestion
            amrex::DeriveRec::TheSameBox,
```

Reply:
Ghosts are set by the interpolation order in ParticleInterpolator. So this would cause an error, unless we allow the user to override the number of ghost zones used.

Reply:
So this is not the number of ghosts in the derived MultiFab (you can set that when you call `Amr::derive` or `AmrLevel::derive`) but rather the number of ghosts you need in the state MultiFab in order to compute the derived quantity. In this case, I don't think you need any.